src/Mvc/Mvc.Core/test/Infrastructure/ValidationProblemDetailsJsonConverterTest.cs

```diff
@@ -135,4 +136,63 @@ public void ReadUsingJsonSerializerWorks()
                 Assert.Equal(new[] { "error1", "error2" }, kvp.Value);
             });
     }
+
+    [Fact]
+    public void WriteWorks()
+    {
+        var problemDetails = new ValidationProblemDetails(new Dictionary<string, string[]>() { { "Property",  new string[]{ "error0" } }})
+        {
+            Title  = "One or more validation errors occurred.",
+            Status = 400
+        };
+
+        var converter = new ValidationProblemDetailsJsonConverter();
+        using MemoryStream stream = new();
+        using Utf8JsonWriter writer = new(stream);
+
+        // Act
+        converter.Write(writer, problemDetails, null);
+
+        writer.Flush();
+        var json = Encoding.UTF8.GetString(stream.ToArray());
+
+        var expectedJSON = $"{{\"title\":\"{problemDetails.Title}\",\"status\":{problemDetails.Status}," +
+            "\"errors\":{\"Property\":[\"error0\"]}}";
+        Assert.NotNull(json);
+        Assert.Equal(expectedJSON, json);
+    }
+
+    [Fact]
+    public void WriteUsingJsonSerializerOptionsWorks()
+    {
+        var errors = new Dictionary<string, string[]>()
+        {
+            { "Property",  new string[]{ "error0" } },
+            { "TwoWords",  new string[]{ "error1" } },
+            { "TopLevelProperty.PropertyName",  new string[]{ "error2" } },
+        };
+        var problemDetails = new ValidationProblemDetails(errors)
+        {
+            Title  = "One or more validation errors occurred.",
+            Status = 400
+        };
+
+        // Act
+        var converter = new ValidationProblemDetailsJsonConverter();
+        using MemoryStream stream = new();
+        using Utf8JsonWriter writer = new(stream);
+
+        var options = new JsonOptions().JsonSerializerOptions;
+        options.DictionaryKeyPolicy = JsonNamingPolicy.CamelCase;
```

Comment:
This should be the default for `JsonOptions` since it uses `JsonSerializerDefaults.Web`, right?

Reply:
I don't think so, based on the source code

https://github.com/dotnet/runtime/blob/27226f631ff5e6019b7da78d350636a6e964aa3c/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L147

it will set the `_jsonPropertyNamingPolicy` but not the `_dictionaryKeyPolicy`